src/Localization/Localization/src/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -153,13 +153,20 @@ public IStringLocalizer Create(Type resourceSource)
                 throw new ArgumentNullException(nameof(resourceSource));
             }
 
-            var typeInfo = resourceSource.GetTypeInfo();
-
-            var baseName = GetResourcePrefix(typeInfo);
+            // Get without Add to prevent unnecessary lambda allocation
+            if (_localizerCache.TryGetValue(resourceSource.AssemblyQualifiedName!, out var cache))
+            {
+                return cache;
+            }
 
-            var assembly = typeInfo.Assembly;
+            return _localizerCache.GetOrAdd(resourceSource.AssemblyQualifiedName!, _ =>
```

Comment:
This lambda has a closure because it needs to get the type. Is there a reason why AssemblyQualifiedName is preferable as a key to the type itself? If the type is the key then it can be accessed in the lambda without a closure and it can be made static.

Reply:
The cache is used by another method that is not using a Type

Reply:
This right https://github.com/dotnet/aspnetcore/blob/a318a56f731c030c7375a33255b13af8310e5703/src/Localization/Localization/src/ResourceManagerStringLocalizerFactory.cs#L190-L197?

Reply:
This change looked simple at first, but now I'm not so sure. The methods before were calling a virtual GetResourcePrefix to resolve the cache key, not that's no longer being called so this is a breaking change if you changed that method expecting to cache based on that key (I don't know if that happens in practice though). One workaround is to only do this change if the exact type is `ResourceManagerStringLocalizerFactory`

Reply:
@pranavkm The `IStringLocalizer` has two `Create` methods, one using a Type, the other using two strings. The current implementation uses a single CD to cache the result of both methods, hence it's using a `string` key. I am not changing that in this PR. 

@davidfowl Before this PR the key was using `GetResourcePrefix(TypeInfo)` directly. Now it's still used in the lambda when the item is added in the cache. This value only differs by type, so by using the type name as the key we are having a direct mapping between the cache entry and the value of this method. The only "breaking change" is that this was called for each cache lookup, and now only once (the goal of this PR since it's reflection heavy).

@JamesNK about the lambda creating a closure, that's why I make a call to the dictionary __without__ the lambda right before. And the type can't be used as the key since the CD is used by the other interface method (that is not using a type).